backend/api/services/groupService.ts

```diff
@@ -254,11 +254,16 @@ async function removeAllUsersFromGroup(cid: number, gid: number) {
  * throws {Error} Throws an error if the network request fails or the response is not ok.
  */
 export async function getUserGroups(uid: number, token: string) {
-    const response = await fetchWithToken(`${apiUrl}/users/${uid}/groups`, token);
-    if (!response.ok){
-        throw new Error(response.statusText);
+    try {
+        const response = await fetchWithToken(`${apiUrl}/users/${uid}/groups`, token);
+        if (!response.ok){
+            throw new Error(response.statusText);
+        }
+        
+        return await response.json() as GroupDetails[];
+    } catch(error) {
+        console.error("Error during fetch", error)
     }
```

Comment:
I would handle the error like this as the prior solution results in an empty 200 response to the client, which should not happened as when an request can not be performed a non 200 status code should be returned. 
```suggestion
      const response = await fetchWithToken(`${apiUrl}/users/${uid}/groups`, token);
      if (!response.ok){
          throw new Error(`Received non 200 status while fetching groups: ${response.statusText}`);
      }
      
      return await response.json() as GroupDetails[];
```

and then catch, log and return the appropriate status code in the controller.

Reply:
this is intended. this api route is called by the Group-Selector to fetch the groups a user can select. If a user has no groups the SQL return is empty. This is not a mistake, as users can have no groups there are cases in which the SQL correctly returns an empty set of data. Sending an error code when the set is empty would imply that an error happend, but returning no groups if the user in in no groups, is no error.